### PR TITLE
Populate Jaeger's Span.Process from Resource

### LINKED ISSUE #1550: Jaeger exporter must populate Jaeger's Span.Process from Resource service.name attribute
Per specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#resource

Should fall back to `resource.Default()` `service.name` attribute if the exported Span does not have its own Resource `service.name`

### PR DESCRIPTION
**Pull request by XSAM**
Resolves #1550

### CODE REVIEW

**Review by evantorrie — COMMENTED** [review]
Looks good! Some comments you may want to look at.

File: exporters/trace/jaeger/jaeger.go [hunk 1]
```diff
@@ -109,27 +102,26 @@ func NewRawExporter(endpointOption EndpointOption, opts ...Option) (*Exporter, e
...
-	for _, tag := range o.Process.Tags {
-		t := keyValueToTag(tag)
-		if t != nil {
-			tags = append(tags, t)
+	// Fetch default service.name from default resource for backup
+	var defaultServiceName attribute.KeyValue
+	defaultResource := resource.Default()
+	for iter := defaultResource.Iter(); iter.Next(); {
+		if iter.Attribute().Key == semconv.ServiceNameKey {
+			defaultServiceName = iter.Attribute()
```
[the remark this answers is not in the document]
Comment (XSAM) [reply exporters/trace/jaeger/jaeger.go:110]: I found the `Resource` has an API to return the instance of `Set`, so I directly use `Set` API to lookup a value now.

File: exporters/trace/jaeger/jaeger.go [hunk 2]
```diff
@@ -425,11 +408,94 @@ func (e *Exporter) Flush() {
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+// jaegerBatchList transforms a slice of SpanSnapshot into a slice of jaeger
+// Batch.
+func jaegerBatchList(ssl []*export.SpanSnapshot, defaultServiceName string, resourceFromProcess *resource.Resource) []*gen.Batch {
```
[the remark this answers is not in the document]
Comment (XSAM) [reply exporters/trace/jaeger/jaeger.go:425]: Yeah, that's correct.

File: exporters/trace/jaeger/jaeger_test.go [hunk 3]
```diff
@@ -327,11 +305,11 @@ func TestExporter_ExportSpan(t *testing.T) {
 	// Create Jaeger Exporter
 	exp, err := NewRawExporter(
 		withTestCollectorEndpoint(),
-		WithProcess(Process{
-			ServiceName: serviceName,
-			Tags: []attribute.KeyValue{
+		WithSDK(&sdktrace.Config{
+			Resource: resource.NewWithAttributes(
+				semconv.ServiceNameKey.String(serviceName),
```
[the remark this answers is not in the document]
Comment (XSAM) [reply exporters/trace/jaeger/jaeger_test.go:310]: > This is something that should probably be added in another PR.

Yeah, it seems we need to refactor the `testCollectorEnpoint`. I created an [issue](#1699) to track this.

File: exporters/trace/jaeger/jaeger_test.go [hunk 4]
```diff
@@ -579,3 +559,226 @@ func TestErrorOnExportShutdownExporter(t *testing.T) {
...
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			batchList := jaegerBatchList(tc.spanSnapshotList, tc.defaultServiceName, tc.resourceFromProcess)
```
[the remark this answers is not in the document]
Comment (XSAM) [reply exporters/trace/jaeger/jaeger_test.go:735]: Oh shoot, I declare the `tc.defaultServiceName` but never use it. 😂

**Review by evantorrie — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**